lib/deref.js

```diff
@@ -86,38 +87,23 @@ module.exports = {
         resolveTo, stack, seenRef, stackLimit);
     }
 
-    // generate one object for each schema
-    let indivObjects = schemaArr.map((schema) => {
-        return this.resolveRefs(schema, parameterSourceOption, components, schemaResolutionCache, resolveFor,
-          resolveTo, stack, seenRef, stackLimit);
-      }).filter((schema) => {
-        return schema.type === 'object';
-      }),
-
-      // generated object with properties from all allOf entries which we return
-      finalObject = {
-        type: 'object',
-        properties: {}
-      },
-
-      // set of properties which we've already handled, to avoid repitition
-      handledProps = {},
-      i,
-      j;
-
-    for (i = 0; i < indivObjects.length; i++) {
-      // go through the indiv props, and add to finalObject if not in handledProps
-      for (j in indivObjects[i].properties) {
-        if (indivObjects[i].properties.hasOwnProperty(j) && !handledProps[j]) {
-          handledProps[j] = true;
-          finalObject.properties[j] = indivObjects[i].properties[j];
+    try {
+      return mergeAllOf({
+        allOf: schemaArr.map((schema) => {
+          return this.resolveRefs(schema, parameterSourceOption, components, schemaResolutionCache, resolveFor,
+            resolveTo, stack, seenRef, stackLimit);
+        })
+      }, {
+        resolvers: {
+          // for keywords in OpenAPI schema that are not standard defined JSON schema keywords, use default resolver
+          defaultResolver: (compacted) => { return compacted[0]; }
         }
-      }
-
-      finalObject.description = finalObject.description || indivObjects[i].description;
+      });
+    }
+    catch (e) {
+      console.warn('Error while resolving allOf schema: ', e);
+      return { 'Error': 'Could not resolve allOf schema' };
```

Comment:
Check the other error messages - `<schema too deep>`. We should follow the same angled-bracket syntax